anesthetic/utils.py

```diff
@@ -89,6 +90,148 @@ def quantile(a, q, w=None, interpolation='linear'):
     return quant
 
 
+def sample_cdf(samples, inverse=False, interpolation='linear'):
+    """Sample the empirical cdf for a 1d array."""
+    samples = np.array(samples)
+    samples.sort()
+    ngaps = len(samples)-1
+    gaps = np.random.dirichlet(np.ones(ngaps))
+    cdf = np.array([0, *np.cumsum(gaps)])
+    assert np.isclose(cdf[-1], 1, atol=1e-9, rtol=1e-9), \
+        "Error: CDF does not reach 1 but "+str(cdf[-1])
+    # Set the last element (tested to be approx 1)
+    # to exactly 1 to avoid interpolation errors
+    cdf[-1] = 1
+    if inverse:
+        return interp1d(cdf, samples, kind=interpolation)
+    else:
+        return interp1d(samples, cdf, kind=interpolation)
+
+
+def credibility_interval(samples, weights=None, level=0.68, method="hpd",
+                         u=None, n_iter=12, return_covariance=False,
+                         verbose=False):
```

Comment:
Good point, updated the docstring!

I would rather not add `interpolation_kwargs` because the interpolation should not matter (thousands of data points). In any case where the interpolation does matter (10-ish points) the method is going to be wrong in any case, and there exists no _good_ interpolation one could choose.

Reply:
I agree that for a lot of samples the interpolation method hardly matters. But a lot of the discussion above was specifically about exploring different interpolation options. So I would suggest to pick your favourite as default, but give the user full flexibility. For one, this would allow the user to at least try different interpolation methods to get a sense of how much that might affect the results.